cmd/agent/app/builder.go

```diff
@@ -74,20 +75,13 @@ type Builder struct {
 	Processors     []ProcessorConfiguration    `yaml:"processors"`
 	SamplingServer SamplingServerConfiguration `yaml:"samplingServer"`
 
-	// CollectorHostPorts are host:ports of a static list of Jaeger Collectors.
-	CollectorHostPorts []string `yaml:"collectorHostPorts"`
-
-	// MinPeers is the min number of servers we want the agent to connect to.
-	// If zero, defaults to min(3, number of peers returned by service discovery)
-	DiscoveryMinPeers int `yaml:"minPeers"`
+	// These 3 fields are copied from reporter.Builder because yaml does not parse embedded structs
+	CollectorHostPorts   []string `yaml:"collectorHostPorts"`
+	DiscoveryMinPeers    int      `yaml:"minPeers"`
+	CollectorServiceName string   `yaml:"collectorServiceName"`
 
-	// CollectorServiceName is the name that Jaeger Collector's TChannel server
-	// responds to.
-	CollectorServiceName string `yaml:"collectorServiceName"`
+	tchreporter.Builder
```

Comment:
> so I did not want to break config compatibility by pulling builder out completely

Is this a very big concern? Do we expect people to roll out newer versions of agents without updating their configs (or having no ability of updating configs in lockstep with the agent)?